custom/icds_reports/utils/topojson_util/README.md

```diff
@@ -0,0 +1,43 @@
+# TopoJson Setup
+
+This module serves topojson files for blocks based on the selected district.
+This avoids having to send down 20+MB of shape data on every dashboard page.
+
+As of December, 2019 this functionality is used by the mobile dashboard and is in QA
+on web.
+
+## Generating the files
+
+The files in `static` have been generated by the management command `split_topojsons` in 
+this application.
+
+You will need to install mapshaper to run it:
+
+```
+npm install -g mapshaper
+```
+
+## File structure
+
+`static/district_topojson_data.json` is a mapping of States to their relevant districts and blocks.
+
+`static/blocks/<state>_blocks_v3.toposon` is the actual block-level topojson data for that 
+state's blocks, keyed by district.
+
+## Functionality
+
+The `get_topojson_for_district` function first looks up the appropriate state file
+for the district by matching on the district name. 
+Then returns the appropriate topojson file for the whole state.
+
+## Todos
+
+The following is a brain dump of todos to improve this setup:
+
+- Reliably run `split_topojsons` and confirm it produces the same results
+- Implement this functionality for districts and not just blocks
+- Don't return the whole state's data when just the single district's blocks are needed
+- Maybe load the data via a static view instead of having it pass through django so it
+  can be served by nginx (would require porting some logic to the front end)
+- Or else, do some caching on the data so we aren't hitting the filesystem with every request
+- Confirm there are no duplcate district names
```

Comment:
Im pretty sure there are definitely duplicate district names in different states (i think it caused a bug in the past). I do not believe there are duplicates in the same state.

Reply:
Ok, I'll figure out how to test for duplicates before releasing this beyond mobile / feature flag.

Reply:
https://github.com/dimagi/commcare-hq/pull/26474